### DISCUSSION TITLE
Custom algorithm in BootstrapODPSample

### DISCUSSION THREAD

**Discussion by DariaThreeTears** [casact/chainladder-python#310] [Q&A] [answered]
Are there any ways to simulate triangles with custom Bootstrap algorithm?
For example, I have set of values (as below) and I want to build triangles with elements from column 'Value', if random number from 0 to 1 is between 'Begin' and 'End'.

Number | Begin     | End          | Value
--           | --            | --             | --
1            | 0             | 0,018182 | 0
2            | 0,018182 | 0,036364 | -481,69
3            | 0,036364 | 0,054545 | -364,741
4            | 0,054545 | 0,072727 | 163,3772
5            | 0,072727 | 0,090909 | 118,7875
6            | 0,090909 | 0,109091 | 181,9839
7            | 0,109091 | 0,127273 | 111,2244

**Comment by jbogaardt** [comment, marked as answer]
Hello @DariaThreeTears, 

I'm not familiar with the technique you're describing.  The BootstrapODPSample is based on [CAS Monograph 4](https://www.casact.org/monograph/cas-monograph-no-4). This is a very specific simulation technique that takes advantage of the fact that the basic chainladder method is equivalent to an ODP GLM regression. It does not provide functionality beyond exploiting those properties. Implicitly, it requires an existing triangle of data.

Perhaps you can elaborate on your use case?

**Reply by DariaThreeTears** [reply]
Ok, thank you
